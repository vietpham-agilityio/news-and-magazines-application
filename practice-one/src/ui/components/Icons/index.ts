export * from '@/ui/components/Icons/User';
export * from '@/ui/components/Icons/Plus';
export * from '@/ui/components/Icons/Home';
export * from '@/ui/components/Icons/File';
export * from '@/ui/components/Icons/Close';
export * from '@/ui/components/Icons/Circle';
export * from '@/ui/components/Icons/Search';
export * from '@/ui/components/Icons/SaveTag';
export * from '@/ui/components/Icons/MailBox';
export * from '@/ui/components/Icons/Twitter';
export * from '@/ui/components/Icons/ChatBox';
export * from '@/ui/components/Icons/Ellipse';
export * from '@/ui/components/Icons/Schedule';
export * from '@/ui/components/Icons/ArrowTop';
export * from '@/ui/components/Icons/ThreeDots';
export * from '@/ui/components/Icons/Triangle';
export * from '@/ui/components/Icons/Instagram';
export * from '@/ui/components/Icons/BurgerMenu';
export * from '@/ui/components/Icons/ChevronsUp';
export * from '@/ui/components/Icons/ArrowLeft';
export * from '@/ui/components/Icons/ArrowBack';
export * from '@/ui/components/Icons/ArrowRight';
export * from '@/ui/components/Icons/PaperPlane';
export * from '@/ui/components/Icons/ArrowBottom';
export * from '@/ui/components/Icons/ChatBoxSmall';
export * from '@/ui/components/Icons/SaveTagSmall';