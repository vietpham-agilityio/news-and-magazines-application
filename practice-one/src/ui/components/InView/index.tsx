'use client';

import { InView } from 'react-intersection-observer';

interface IProps {
  children: React.ReactNode;
}

const InViewWrapper = ({ children }: IProps) => {
  return (
    <InView triggerOnce threshold={1}>
      {({ ref, inView }) => (
        <div ref={ref} className="w-full">
          {inView && children}
        </div>
      )}
    </InView>
  );
};

export default InViewWrapper;
