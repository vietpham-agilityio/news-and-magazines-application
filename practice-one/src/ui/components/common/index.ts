export * from '@/ui/components/common/Tag';
export * from '@/ui/components/common/Menu';
export * from '@/ui/components/common/Title';
export * from '@/ui/components/common/Input';
export * from '@/ui/components/common/Avatar';
export * from '@/ui/components/common/Button';
export * from '@/ui/components/common/BoxIcon';
export * from '@/ui/components/common/SubMenu';
export * from '@/ui/components/common/TextArea';
export * from '@/ui/components/common/Typography';
export * from '@/ui/components/common/Information';
