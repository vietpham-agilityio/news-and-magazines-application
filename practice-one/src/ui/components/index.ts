// common
export * from '@/ui/components/common';

// shared
export * from '@/ui/components/Brand';
export * from '@/ui/components/NavItem';
export * from '@/ui/components/Comment';
export * from '@/ui/components/Carousel';
export * from '@/ui/components/SearchBar';
export * from '@/ui/components/PaperPost';
export * from '@/ui/components/BurgerMenu';
export * from '@/ui/components/Pagination';
export * from '@/ui/components/CommentForm';
export * from '@/ui/components/BreadCrumbs';
export * from '@/ui/components/TitleSection';
export * from '@/ui/components/SubmitButton';
export * from '@/ui/components/Card/CardSingle';
export * from '@/ui/components/Skeleton/Comment';
export * from '@/ui/components/Skeleton/CardPost';
export * from '@/ui/components/Skeleton/BadgeGroup';
export * from '@/ui/components/Skeleton/DetailPost';
export * from '@/ui/components/Skeleton/CardVertical';
export * from '@/ui/components/Skeleton/CardCategory';
export { default as CardPost } from '@/ui/components/Card/CardPost';
export { default as CardCategory } from '@/ui/components/Card/CardCategory';