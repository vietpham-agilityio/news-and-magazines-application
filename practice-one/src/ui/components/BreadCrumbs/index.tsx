// components
import Link from 'next/link';
import { Typography } from '@/ui/components';

// types
import { FontWeight, IBreadCrumbItem, Size } from '@/types';

// icons
import { ArrowRight } from '@/ui/components/Icons';

interface IProps {
  listBreadCrumb: IBreadCrumbItem[];
}

export const BreadCrumbs = ({ listBreadCrumb }: IProps) => (
  <div className="flex items-center">
    {listBreadCrumb.map((item, index) => {
      const { path, name } = item;

      return (
        <div key={name} className="flex items-center">
          <Link href={path}>
            <Typography
              tag="h2"
              textSize={Size.XS}
              weight={FontWeight.Normal}
              additionalClasses={`
              ${
                index === listBreadCrumb.length - 1
                  ? 'w-57.5 text-dark-90 h-5 truncate text-ellipsis overflow-hidden sm:w-full'
                  : ' text-dark-100 hover:text-primary-100'
              }`}
            >
              {name}
            </Typography>
          </Link>
          {index !== listBreadCrumb.length - 1 && (
            <i className="mx-2">
              <ArrowRight />
            </i>
          )}
        </div>
      );
    })}
  </div>
);
