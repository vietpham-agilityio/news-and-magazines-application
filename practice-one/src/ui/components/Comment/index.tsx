// static image
import ImageStore from '@/constants/images';

// components
import { Avatar, Button, Typography } from '@/ui/components';

// icons
import { ArrowBack, Schedule } from '@/ui/components/Icons';

// types
import { FontWeight, Size } from '@/types';

interface IProps {
  userName: string;
  content: string;
  publicDate?: string;
}

export const Comment = ({ userName, content, publicDate }: IProps) => (
  <div className="bg-white-90 p-4 rounded-xl">
    { publicDate ? (
      <div className="flex flex-1 px-1 mb-4">
        <Avatar
          width={60}
          height={60}
          url={ImageStore.AvatarUser}
          alt="Author this comment"
        />
        <div className="flex flex-1 items-center justify-between">
          <div className="flex flex-1 justify-between">
            <div className="flex flex-col h-full ml-2.5">
              <Typography
                tag="h2"
                textSize={Size.SM}
                additionalClasses="text-dark-100"
              >
                {userName}
              </Typography>
              <div className="flex gap-2">
                <Schedule />
                <Typography
                  tag="h3"
                  weight={FontWeight.Medium}
                  additionalClasses="text-dark-75 text-p leading-5"
                >
                  {publicDate}
                </Typography>
              </div>
            </div>
          </div>
        </div>
      </div>
    ) : (
      <Typography tag="h2" textSize={Size.SM} additionalClasses="text-dark-100">
        {userName}
      </Typography>
    )}
    <Typography
      tag="h3"
      weight={FontWeight.Light}
      additionalClasses={`${publicDate ? 'text-xsm text-black-75 leading-5' : 'text-p text-dark-75'}`}
    >
      {content}
    </Typography>
  </div>
);
