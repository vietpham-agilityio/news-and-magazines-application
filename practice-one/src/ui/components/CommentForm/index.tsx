// components
import { createComment } from '@/services';
import { Input, SubmitButton, TextArea } from '@/ui/components';

// icons
import { ChatBoxSmall } from '@/ui/components/Icons';

interface IProps {
  postId: string;
}

export const CommentForm = ({postId}: IProps) => {

return (
  <form action={createComment}>
    <div className="grid grid-cols-12 grid-rows-3 lg:gap-6">
      <div className="row-span-3 col-span-12 lg:col-span-6">
        <input name="postId" type="hidden" value={postId} />
        <div className="mb-5 row-span-1">
          <Input label="Name" name="name" type="text" />
        </div>
        <div className="mb-5 row-span-1">
          <Input label="Email" name="email" type="email" />
        </div>
        <div className="mb-5 row-span-1">
          <Input label="Website" name="website" type="text" />
        </div>
      </div>
      <div className="row-span-3 col-span-12 lg:col-span-6">
        <TextArea name='content' label="Comment" placeholder="What do you think?" />
      </div>
    </div>
    <div className="flex w-full justify-end mt-7 mb-30 sm:mt-10 sm:mb-12 lg:mt-0">
      <SubmitButton
        label="Send Comment"
        leftIcon={<ChatBoxSmall />}
      />
    </div>
  </form>
)}
