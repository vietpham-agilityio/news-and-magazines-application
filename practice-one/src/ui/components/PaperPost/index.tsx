import { StaticImageData } from 'next/image';
import { LazyImage, Typography } from '@/ui/components';

// types
import { FontWeight, Size } from '@/types';
import Link from 'next/link';

interface IProps {
  id: number;
  title: string;
  content: string;
  imageSrc: string | StaticImageData;
  alt: string;
  isLargePaper?: boolean;
}

export const PaperPost = ({
  id,
  title,
  content,
  imageSrc,
  alt,
  isLargePaper,
}: IProps) => {
  const titleSize = isLargePaper ? Size.SM : Size.XS;
  const contentSize = isLargePaper ? 'text-xsm' : 'text-p';
  const imageWidth = isLargePaper ? '190px' : '87px';
  const imageHeight = isLargePaper ? '190px' : '87px';

  return (
    <Link href={`/articles/${id}`}>
      <div
        className={`flex cursor-pointer ${isLargePaper && 'bg-white-100 h-52.5 w-full p-2.5 rounded-xl'}`}
      >
        <div
          style={{ width: imageWidth, height: imageHeight, position: 'relative' }}
        >
          <LazyImage
            fill
            objectFit="cover"
            additionalClasses="rounded-xl"
            src={imageSrc}
            alt={alt}
            sizes={`${isLargePaper ? '190px' : '87px'}`}
          />
        </div>
        <div
          className={`flex flex-col gap-1.5 ml-2.5 flex-1 ${isLargePaper ? 'py-4' : ' my-3'}`}
        >
          <Typography
            tag="h2"
            textSize={titleSize}
            additionalClasses={`w-[223px] text-dark-100 text-ellipsis  ${isLargePaper && 'h-7 px-1.5 w-auto'}`}
          >
            {title}
          </Typography>
          <Typography
            tag="h3"
            weight={FontWeight.Light}
            additionalClasses={`
            h-8 w-[223px] text-dark-75 text-wrap truncate text-ellipsis overflow-hidden
            ${contentSize}
            ${isLargePaper && 'px-1.5 h-auto w-auto'}`}
          >
            {content}
          </Typography>
        </div>
      </div>
    </Link>
  );
};
