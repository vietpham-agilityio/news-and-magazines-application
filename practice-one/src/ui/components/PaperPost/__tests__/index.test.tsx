import { render, cleanup } from '@testing-library/react';

// component
import { PaperPost } from '@/ui/components';

let renderPaperPost: any;
const image: string =
  'https://images.unsplash.com/photo-1563026915-32eba3e315de?q=80&w=2531&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D';
const altValue: string = 'This is museum';
const title: string = 'The Privs museum close on 5AM';
const content: string = 'The Privs start close and maintain all of the wall';

describe('PaperPost component', () => {
  beforeEach(() => {
    renderPaperPost = render(
      <PaperPost
        id={1}
        isLargePaper
        imageSrc={image}
        title={title}
        alt={altValue}
        content={content}
      />
    );
  });

  afterEach(() => {
    renderPaperPost.unmount();
    cleanup();
  });

  it('should render PaperPost macth snapshot', () => {
    expect(renderPaperPost.asFragment()).toMatchSnapshot();
  });

  it('should render PaperPost without isLargePaper prop to macth snapshot', () => {
    const { asFragment } = render(
      <PaperPost
        id={1}
        imageSrc={image}
        title={title}
        alt={altValue}
        content={content}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('PaperPost should render value passed title props', () => {
    const { getByText } = renderPaperPost;

    const titleElement = getByText(title);

    expect(titleElement).toBeInTheDocument();
  });

  it('PaperPost should render value passed content props', () => {
    const { getByText } = renderPaperPost;

    const contentElement = getByText(content);

    expect(contentElement).toBeInTheDocument();
  });

  it('PaperPost should render value passed alt props', () => {
    const { getByAltText } = renderPaperPost;

    const imageElement = getByAltText(altValue);

    expect(imageElement).toBeInTheDocument();
    expect(imageElement).toHaveAttribute('alt', 'This is museum');
  });
});
