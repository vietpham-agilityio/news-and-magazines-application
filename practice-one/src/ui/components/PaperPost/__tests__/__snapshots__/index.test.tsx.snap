// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaperPost component should render PaperPost macth snapshot 1`] = `
<DocumentFragment>
  <a
    href="/articles/1"
  >
    <div
      class="flex cursor-pointer bg-white-100 h-52.5 w-full p-2.5 rounded-xl"
    >
      <div
        style="width: 190px; height: 190px; position: relative;"
      >
        <img
          alt="This is museum"
          class="rounded-xl"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="190px"
          src="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=3840&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=16&q=75 16w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=32&q=75 32w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=48&q=75 48w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=64&q=75 64w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=96&q=75 96w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=128&q=75 128w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
        />
      </div>
      <div
        class="flex flex-col gap-1.5 ml-2.5 flex-1 py-4"
      >
        <h2
          aria-level="2"
          class="text-sm font-medium w-[223px] text-dark-100 text-ellipsis  h-7 px-1.5 w-auto"
          role="heading"
        >
          The Privs museum close on 5AM
        </h2>
        <h3
          aria-level="3"
          class="text-md font-light 
            h-8 w-[223px] text-dark-75 text-wrap truncate text-ellipsis overflow-hidden
            text-xsm
            px-1.5 h-auto w-auto"
          role="heading"
        >
          The Privs start close and maintain all of the wall
        </h3>
      </div>
    </div>
  </a>
</DocumentFragment>
`;

exports[`PaperPost component should render PaperPost without isLargePaper prop to macth snapshot 1`] = `
<DocumentFragment>
  <a
    href="/articles/1"
  >
    <div
      class="flex cursor-pointer undefined"
    >
      <div
        style="width: 87px; height: 87px; position: relative;"
      >
        <img
          alt="This is museum"
          class="rounded-xl"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="87px"
          src="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=3840&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=16&q=75 16w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=32&q=75 32w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=48&q=75 48w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=64&q=75 64w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=96&q=75 96w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=128&q=75 128w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1563026915-32eba3e315de%3Fq%3D80%26w%3D2531%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
        />
      </div>
      <div
        class="flex flex-col gap-1.5 ml-2.5 flex-1  my-3"
      >
        <h2
          aria-level="2"
          class="text-xs font-medium w-[223px] text-dark-100 text-ellipsis  undefined"
          role="heading"
        >
          The Privs museum close on 5AM
        </h2>
        <h3
          aria-level="3"
          class="text-md font-light 
            h-8 w-[223px] text-dark-75 text-wrap truncate text-ellipsis overflow-hidden
            text-p
            undefined"
          role="heading"
        >
          The Privs start close and maintain all of the wall
        </h3>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
